src/controllers/planterRegistration.controller.ts

```diff
@@ -22,28 +15,44 @@ export class PlanterRegistrationController {
         description: 'Array of PlanterRegistration model instances',
         content: {
           'application/json': {
-            schema: {type: 'array', items: {'x-ts-type': PlanterRegistration}},
+            schema: {
+              type: 'array',
+              items: { 'x-ts-type': PlanterRegistration },
+            },
           },
         },
       },
     },
   })
   async find(
-    @param.query.object('filter', getFilterSchemaFor(PlanterRegistration))
-    filter?: Filter<PlanterRegistration>,
+    @param.query.object('filter') filter: object,
   ): Promise<PlanterRegistration[]> {
-    return await this.planterRepository.find(filter);
```

Comment:
I'm a little concerned about removing the full support for `filter`.
If someone queries this endpoint with parameters other than `where.planterId`, they will be silently ignored.

There's a slightly convoluted example in [`trees.controller.js`](https://github.com/Greenstand/treetracker-admin-api/blob/46583de2ff47d3d0914e9de6a53a7e26894cc3ef/src/controllers/trees.controller.ts#L273) of how we can augment a loopback query.

You'd just need to do the `where` bits to build the `WHERE` clause.

Reply:
I tried to add support for `filter` as suggested. Please let me know if this is the right way to do it.
I also abstracted `buildFilterQuery` to its own file and replaced it in `trees.controller.ts` so we don't repeat ourselves.

Thanks for the feedback!

Reply:
Awesome! Nice work.